Rakefile

```diff
@@ -1,24 +1,19 @@
 require 'rake'
 require 'rake/testtask'
 
+require_relative 'lib/tasks/exercise'
+require_relative 'lib/tasks/exercise_test_tasks'
 require_relative 'lib/tasks/exercise_tests_runner'
-require_relative 'lib/tasks/knows_what_exercises_are_called_and_where_they_live_in_the_directory_structure'
+require_relative 'lib/tasks/rake_arguments'
 
-Exercise.all.each do |exercise|
-  task exercise do
-    ExerciseTestsRunner.new(
-      exercise: exercise,
-      test_options: ARGV.drop_while { |e| e != '--' }.drop(1).join(' ')
-    ).run
-  end
-end
+ExerciseTestTasks.new exercises: Exercise.all,
+                      options: RakeArguments.task_options
 
 task default: 'test'
 
 desc 'Run individual exercises or run all development and exercise tests'
 task :test do
-  exercises = ARGV.take_while { |e| e != '--' }.drop(1)
-  next if exercises.any?
+  next if RakeArguments.more_tasks?
```

Comment:
The non-standard thing we are doing here is *ignoring* the test option if it comes first.

I'm beginning to think that we should not be doing that.